### PR TITLE
tracing of requests & internals

### PR DESCRIPTION
**Pull request by danielnorberg**
add tracing to styx api/scheduler requests and internals

A next step after this PR is to instrument the storage methods etc with spans to get more granular breakdown of what's going on in there.

### CODE REVIEW

File: styx-common/src/main/java/com/spotify/styx/api/Middlewares.java [hunk 1]
```diff
@@ -220,6 +226,33 @@ static GoogleIdToken verifyIdToken(String s, GoogleIdTokenVerifier verifier) {
...
+        span.setStatus(UNKNOWN);
+        span.end();
+        Throwables.throwIfUnchecked(e);
+        throw new RuntimeException(e);
+      }
+      response.whenComplete((r, ex) -> {
+        if (ex != null) {
+          span.setStatus(UNKNOWN);
+        }
+        span.end();
```
Comment (ulzha) [inline styx-common/src/main/java/com/spotify/styx/api/Middlewares.java]: Doing `end()` in multiple places. Consider try-with-resource instead? https://static.javadoc.io/io.opencensus/opencensus-api/0.13.2/io/opencensus/trace/SpanBuilder.html#startScopedSpan--

File: styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java [hunk 2]
```diff
@@ -420,7 +425,7 @@ public void updateNextNaturalTrigger(WorkflowId workflowId, TriggerInstantSpec t
 
     // Strongly consistently read values for the above keys
     return Lists.partition(keys, MAX_NUMBER_OF_ENTITIES_IN_ONE_BATCH_READ).stream()
-        .map(batch -> forkJoinPool.submit(withMDC(() -> this.readRunStateBatch(batch))))
+        .map(batch -> CompletableFuture.supplyAsync(() -> this.readRunStateBatch(batch)))
```
Comment (ulzha) [inline styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java]: `executor` not passed, by accident?

File: styx-common/src/main/java/com/spotify/styx/api/Middlewares.java (hunk 1 above)
Comment (danielnorberg) [reply styx-common/src/main/java/com/spotify/styx/api/Middlewares.java]: The request handler might be async, in which case it returns a future and we'd end up closing the span before it's (asynchronously) completed.

File: styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java (hunk 2 above)
Comment (danielnorberg) [reply styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java]: good catch!

File: styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java [hunk 3]
```diff
@@ -420,7 +425,7 @@ public void updateNextNaturalTrigger(WorkflowId workflowId, TriggerInstantSpec t
 
     // Strongly consistently read values for the above keys
     return Lists.partition(keys, MAX_NUMBER_OF_ENTITIES_IN_ONE_BATCH_READ).stream()
-        .map(batch -> forkJoinPool.submit(withMDC(() -> this.readRunStateBatch(batch))))
+        .map(batch -> CompletableFuture.supplyAsync(() -> this.readRunStateBatch(batch), forkJoinPool))
```
Comment (ulzha) [inline styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java]: Erm I think you meant `executor`, like in the other replaced usages of `withMDC`.

Comment (danielnorberg) [reply styx-common/src/main/java/com/spotify/styx/storage/DatastoreStorage.java]: Good catch again!

**Pull request merged**